### DISCUSSION TITLE
October 2023 Update Report

### DISCUSSION THREAD

**Discussion by MotoLegacy** [nzp-team/nzportable#416] [Announcements]
### Introduction
This post is also available on [ModDB](https://www.moddb.com/games/nazi-zombies-portable/news/october-2023-update-report).

It's that time again! I hope everyone who celebrated enjoyed their Halloween, and that those who have been keeping up to date on NZ:P news through the GitHub stream have felt satisfied with progress. There were a lot of changes made across all platforms this month, and as always this post will give you the run-down on the most important.

As always, consider becoming a [Patron](https://www.patreon.com/CypressImplex). I'm able to allocate as much time as I do nowadays thanks to that donation stream.

October saw 103 commits across all project modules. Not bad!

### Significant Nintendo 3DS Overhauls

![IMG_6667](https://github.com/nzp-team/nzportable/assets/26030250/3a1e6fc8-af38-47de-baaa-e7aeb6a850e7)

One of the biggest changes this month was a primary focus in the Nintendo 3DS client. While still lacking in some regards such as performance (though performance *has* improved), a lot of missing elements such as skyboxes, alias model lighting, various piece of UI, etc. have been re-instated from other platforms. If you were previously disappointed with the Nintendo 3DS builds of NZ:P, I'd recommend you check it out again on [Universal Updater](https://db.universal-team.net/3ds/nazi-zombies-portable), and report any issues you still see present. 

### FTE (PC) Particle Enhancements

https://github.com/nzp-team/nzportable/assets/26030250/23de4c97-82ee-433a-b6e9-5a8031f2db59

The FTE builds of NZ:P had a lot weaker (and less performant) particles than other builds, this has mostly been addressed, with faster and more accurate particles for explosions, blood, and weapon traces. An issue was also addressed where the explosion sound played was less impactful than the one on other platforms.

### Mustang & Sally Revamp

https://github.com/nzp-team/nzportable/assets/26030250/750f651e-05c3-4c27-8928-b874c826b06d

To coincide with the fixed FTE particles, the Mustang & Sally also got a new model and set of animations, making the weapon a lot more fun to use and experiment with!

### Dynamic Crosshairs

https://github.com/nzp-team/nzportable/assets/26030250/b844398f-2cf9-4377-8d61-1cacaaebc7ca

Crosshairs now respond to actions performed, such as reloading, cooking a grenade, or meleeing.

### Bouncing Betty Instructional Text

https://github.com/nzp-team/nzportable/assets/26030250/b6cbdc46-b15a-4afe-b174-a21c1a5b9aee

An accessibility feature brought over from the source material, purchasing a Bouncing Betty now gives valuable information on usage. This was particularly useful on platforms with limited control schemes (a-la, PlayStation Portable) where instructions were necessary to know double-tapping +USE performed an active equipment swap.

### New Map: "House" from LEXX

![Base_Profile_Screenshot_2023 10 12_-_22 28 31 09](https://github.com/nzp-team/nzportable/assets/26030250/10114d16-dee3-4653-8b5d-3861cc315ff0)

I've collaborated with LEXX to bring the Call of Duty: Black Ops (Nintendo DS) map _HOUSE_ to NZ:P, with new secrets and a nicely refined visual style (yes, I repurposed the _TEMPLE_ text from last month, it does the job nicely).

### Improvements to Maximum Zombie Count

Formerly 12 across all non-FTE (PC) platforms, different host servers can now control and determine the maximum amount of Zombies allowed alive at a time when the match starts via the new `nzp_maxai` built-in. Here are the new numbers:

* Nintendo Switch/PlayStation VITA + FTE: 24 (parity)
* NEW Nintendo 3DS: 18
* OLD Nintendo 3DS/PlayStation Portable: 12 (unchanged)

This will be a nice improvement for most platforms, allowing for an increase in difficulty.

### Weapon Inventory Overhaul / Mule Kick fixes

The way clients store weapons internally has been completely revamped, allowing for easier (and as such, improved) internal tracking for a third (Mule Kick) weapon. NZ:P no longer gets confused with weapon order when acquiring and swapping between a third weapon, meaning you can safely put a weapon in your third slot and feel confident it isn't going to replace one of the other two when facing a down.

### Directional-dependent Item triggering

https://github.com/nzp-team/nzportable/assets/26030250/3a63b2f1-ca4b-4e92-a1d5-3470e48e77e1

Another small but difficulty-impacting change is that several items such as Perk-A-Colas and Wall Chalk now require the player to be directly facing them in order for them to be triggered, in parity with the source material. This will fix "exploits" where, for example, a player could stand in front of a wall weapon indefinitely, firing forwards, and purchasing ammo without needing to look away from any potential targets.

### Improved Support for "Legacy" Maps

Something a little harder to illustrate is how NZ:P now has improved support for maps made in the "Demo" and "Unofficial Patch" eras of the game. This includes various changes such as the game now algorithmically providing support for cooperative play spawns with maps that only have one spawn point, or changes to features like radios to fix playback on those old maps. NZ:P now has smarter detection for re-directing paths for props, addressing a lot of issues where the game couldn't load in various models, or specifically when old lamp sprites were offset above their intended origin by 32 units.

### Misc. Bug Fix/Small Addition Bullet List

**FTE (PC):**

- `TryWalkToEnemy` now ensures their full body can reach a goal entity before attempting to ignore waypoints and walk directly towards it.
- Muzzleflashes now correctly follow a weapon when Aiming down the Sights, and responds appropriately to weapon recoil.
- Fixed an issue where a legacy effect would still trigger when impacting an enemy, leading to untextured planes to draw at the world origin.

**dQuakePlus (PSP):**

- Fog now blends into sky boxes allowing for a smoother transition.
- Fixed an issue where `chapertitle` text at the start of the level would clip with other text on the HUD.
- Fixed an issue where .MP3 music streams would always forcefully loop.

**Global:**

- Fixed a parity issue where player would not be granted a generous clip of ammunition if they had none when entering Last Stand.
- Players now "put away" their weapon before starting to drink a Perk-A-Cola.
- Fixed an issue where dual wield weapons wouldn't properly display a left-sided muzzle flash.
- Fixed an issue where performing actions with a right-handed weapon would delay actions with a left-handed weapon.
- Fixed an issue where if the host acquires the Wunderwaffe DG-2 from the Mystery Box, other clients could still acquire it for themselves.
- Fixed an issue where the generic Ammo Box entity could be purchased indefinitely.
- Fixed an issue where the Mystery Box glow's frame would not reset back to 0 when the Mystery Box moved.
- Restored the random Power-Up drop rate from 4% back down to 2%.
- Fixed an issue where the Ray Gun's damage output was the same as a Grenade.
- Fixed an issue where `ching.wav` was not being precached by Doors and Debris.

And more, and more, and more.. See you next month!

**Thanks to my [Patrons](https://www.patreon.com/CypressImplex)!**

- K Z
- mochi
- Rinse a Gateau Mandate
- Tyler Young
- Benjamin Engelhardt
- botdog
- Derped Crusader
- Jackson Clayton
- Ryan Baldwin
- LEXX

**Comment by 0xLLUUKKEE** [comment]
Thank you for all of your hard work!

**Comment by CrisOvO** [comment]
Hello.. I have just installed the 3ds version on my old 3ds xl, the problem I see is that it lacks the way to operate several buttons (Aim, Reload, Pick Up Weapon, Throw Grenade, etc) so it makes the game very difficult to play

**Reply by StickyPuddingg** [reply]
All the controls are configurable in the menu if you are having trouble selecting multiple at once. I play on new 2ds and have no issues with this. 

Now if you mean you cant walk forward and reload simultaneously. I wouldnt understand what the issue is and i recommend you uninstall and reinstall the game. If that dont work. Try troubleshooting the issue yourself, and get back to me if anything changes.
